### PR TITLE
New Addon: Remix tree button on project pages

### PR DESCRIPTION
**Pull request by GDUcrash**
Adds a remix tree button to project pages. The default color is blue, but it lets you choose a custom one.

### CODE REVIEW

**Comment by WorldLanguages** [comment]
@GDUcrash Please use a setting of type `"color"`

**Comment by GDUcrash** [comment]
ok, i was about to ask if there's a "color" settings type. Will fix tomorrow

**Comment by GDUcrash** [comment]
screw it, gonna do it today

**Comment by GDUcrash** [comment]
@WorldLanguages

File: addons/remix-tree-button/main.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+export default async function ({ addon, global, console }) {
+  //define remix tree button elements
+
+  const subactions = document.querySelector(".flex-row.subactions").querySelector(".action-buttons");
+
+  const remixtree = document.createElement("button");
+  remixtree.className = "button action-button remixtree-button";
+  remixtree.id = "scratchAddonsRemixTreeBtn";
+  remixtree.innerHTML = `
```
Comment (WorldLanguages) [inline addons/remix-tree-button/main.js:9]: Don't use innerHTML

File: addons/remix-tree-button/main.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <img src="https://scratch.mit.edu/svgs/project/remix-white.svg" height="15px" style="margin-right: 5px;" />
+    <span>Remix Tree</span>
+    `;
+  remixtree.onclick = () => {
```
Comment (WorldLanguages) [inline addons/remix-tree-button/main.js:13]: `addEventListener` is preferred

File: addons/remix-tree-button/main.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    if (window.location.href.endsWith("/")) window.location.href += "remixtree";
```
Comment (WorldLanguages) [inline addons/remix-tree-button/main.js:14]: You probably want to check whether the URL has a hash? I don't think the button would work if it has one, for example, `#comments-12345`

File: addons/remix-tree-button/main.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    else window.location.href += "/remixtree";
+  };
+  remixtree.style.display = "flex";
+  remixtree.style.alignItems = "center";
+  if (addon.settings.get("buttonColor")) {
+    remixtree.style.backgroundColor = hexTest(addon.settings.get("buttonColor"));
```
Comment (WorldLanguages) [inline addons/remix-tree-button/main.js:20]: There's no need to check whether it's valid hex. Scratch Addons does that for you

File: addons/remix-tree-button/main.js (hunk 1 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js:9]: well it's much more efficient than creating separate image and text elements as variables

File: addons/remix-tree-button/main.js (hunk 2 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js:13]: aren't they the same thing? except .onclick is shorter

File: addons/remix-tree-button/main.js (hunk 3 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js:14]: ok, will fix that

File: addons/remix-tree-button/main.js (hunk 4 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js:20]: ok

File: addons/remix-tree-button/main.js (hunk 1 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js]: fixed

File: addons/remix-tree-button/main.js (hunk 2 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js]: fixed

File: addons/remix-tree-button/main.js (hunk 3 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js]: fixed

File: addons/remix-tree-button/main.js (hunk 4 above)
Comment (GDUcrash) [reply addons/remix-tree-button/main.js]: fixed

**Comment by GDUcrash** [comment]
@WorldLanguages done :D

**Pull request merged**